packages/atomic/src/components/search/atomic-pager/atomic-pager.spec.ts

```diff
@@ -0,0 +1,497 @@
+
+    await expect.element(parts.buttons!).toBeInTheDocument();
+    await expect.element(parts.pageButtons!).toBeInTheDocument();
+    await expect.element(parts.pageButton!).toBeInTheDocument();
+    await expect.element(parts.previousButton!).toBeInTheDocument();
+    await expect.element(parts.nextButton!).toBeInTheDocument();
+    await expect.element(parts.previousButtonIcon!).toBeInTheDocument();
+    await expect.element(parts.nextButtonIcon!).toBeInTheDocument();
+  });
+
+  describe('when pager state changes', () => {
+    it('should update currentPages when pager state changes', async () => {
+      const element = await renderPager({
+        pagerState: {currentPages: [6, 7, 8, 9, 10]},
+      });
+
+      expect(element.pagerState.currentPages).toEqual([6, 7, 8, 9, 10]);
+    });
+
+    it('should update hasPreviousPage when pager state changes', async () => {
+      const element = await renderPager({
+        pagerState: {hasPreviousPage: false},
+      });
+
+      expect(element.pagerState.hasPreviousPage).toBe(false);
+    });
+
+    it('should update hasNextPage when pager state changes', async () => {
+      const element = await renderPager({
+        pagerState: {hasNextPage: false},
+      });
+
+      expect(element.pagerState.hasNextPage).toBe(false);
+    });
+  });
+
+  describe('when search status state changes', () => {
+    it('should update hasError when search status changes', async () => {
+      const element = await renderPager({
+        searchStatusState: {hasError: true},
+      });
+
+      expect(element.searchStatusState.hasError).toBe(true);
+    });
+
+    it('should update hasResults when search status changes', async () => {
+      const element = await renderPager({
+        searchStatusState: {hasResults: false},
+      });
+
+      expect(element.searchStatusState.hasResults).toBe(false);
+    });
+  });
+
+  describe('#validateProps', () => {
+    it('should allow numberOfPages of 0', async () => {
+      const consoleErrorSpy = vi
+        .spyOn(console, 'error')
+        .mockImplementation(() => {});
+
+      await renderPager({numberOfPages: 0});
+
+      expect(consoleErrorSpy).not.toHaveBeenCalled();
+    });
+
+    it('should handle numberOfPages as string number', async () => {
+      const element = await renderPager();
+
+      element.setAttribute('number-of-pages', '8');
+      await element.updateComplete;
+
+      expect(element.numberOfPages).toBe(8);
+    });
+
+    it('should parse numberOfPages from mixed string-number input', async () => {
+      const element = await renderPager();
+
+      const mixedValue = parseInt('9k3', 10);
+      element.numberOfPages = mixedValue;
+      await element.updateComplete;
+
+      expect(element.numberOfPages).toBe(9);
+    });
+
+    it('should handle invalid numberOfPages gracefully', async () => {
+      const element = await renderPager();
+
+      element.numberOfPages = -5;
+
+      expect(() => {
```

Comment:
Don't call validateProps directly, because it's private; rather test that renderPager will throw.